.github/workflows/build-and-test.yml

```
@@ -0,0 +1,116 @@
+name: "Build and Test"
+
+on:
+    push:
+        branches: [master]
+    pull_request:
+        branches: [master]
+    workflow_dispatch:
+
+jobs:
+    unit-tests:
+        name: ${{ matrix.framework }} on ${{ matrix.os }}
+        runs-on: ${{ matrix.os }}
+        permissions:
+            actions: read
+            contents: read
+        strategy:
+            fail-fast: false
+            matrix:
+                # list all OS and frameworks here, then use "exclude" below to skip some combinations
+                # and "include" to add the "sdk_version" mapping
+                os: [ubuntu-latest, windows-latest, macos-latest]
+                framework: [netcoreapp2.1, netcoreapp3.0, netcoreapp3.1, net5.0, net6.0, net7.0, net8.0, net9.0, net10.0, net48]
+                exclude:
+                    # net48 only runs on Windows: exclude in from Linux and macOS
+                    - os: ubuntu-latest
+                      framework: net48
+                    - os: macos-latest
+                      framework: net48
+                    # .NET SDK macOS on arm64 (Apple Silicon) is only supported in net6.0 and above: exclude older frameworks
+                    - os: macos-latest
+                      framework: netcoreapp2.1
+                    - os: macos-latest
+                      framework: netcoreapp3.0
+                    - os: macos-latest
+                      framework: netcoreapp3.1
+                    - os: macos-latest
+                      framework: net5.0
+                # map the target framework (TFM) to the .NET version.
+                # we always install and use .NET SDK 10 for building everything, but we also need the matching runtime for running the tests.
+                # (we only need the runtimes, not the full SDK, but `actions/setup-dotnet` can only install SDK, which includes the runtime)
+                include:
+                    - framework: netcoreapp2.1
+                      sdk_version: 2.1.x
+                    - framework: netcoreapp3.0
+                      sdk_version: 3.0.x
+                    - framework: netcoreapp3.1
+                      sdk_version: 3.1.x
+                    - framework: net5.0
+                      sdk_version: 5.0.x
+                    - framework: net6.0
+                      sdk_version: 6.0.x
+                    - framework: net7.0
+                      sdk_version: 7.0.x
+                    - framework: net8.0
+                      sdk_version: 8.0.x
+                    - framework: net9.0
+                      sdk_version: 9.0.x
+                    - framework: net10.0
+                      sdk_version: 10.0.x
+                    - framework: net48
+                      sdk_version: 10.0.x
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
```

Always use sha pinning. Can probably use shallow checkout too (can remove if it doesn't work)
```suggestion
            - name: Checkout repository
              uses: actions/checkout@1af3b93b6815bc44a9784bd300feb67ff0d1eeb3 # v6.0.0
              with:
                fetch-depth: 0
```